docs/user_guide.md

```diff
@@ -0,0 +1,143 @@
+
+The `display_df` function is designed to provide an interactive tabular display of DataFrames within Jupyter Notebooks. 
+It leverages the [itables](https://github.com/mwouts/itables) extension, which enhances the display of data tables by providing features like sorting, 
+filtering, and pagination directly in the notebook interface.
+
+##### Example Usage:
+```python
+display_df(spark.sql(f"SELECT * FROM {namespace}.annotation"))
+```
+![display_df](screen_shots/display_func.png)
+
+### 5. Accessing Data:
+
+#### 5.1. Showing Available Namespaces and Listing Tables:
+To list all namespaces (databases) and display the tables within each namespace, you can use the following code snippet:
+
+```python
+namespaces = spark.sql("SHOW DATABASES").collect()
+
+# List tables within each namespace
+for namespace in namespaces:
+    namespace_name = namespace.namespace
+    print(f"Tables in namespace: {namespace_name}")
+    spark.sql(f"SHOW TABLES IN {namespace_name}").show(50, truncate=False)
+```
+
+### 6. Closing the Spark Session:
+Please remember to close the Spark session when you are done with your work. This will release the resources and 
+prevent any memory leaks.
+
+```python
+spark.stop()
+```
+
+Please be aware that your session will automatically close after `1 hour`. Should you require an extension, simply invoke 
+`get_spark_session()` to initiate a new session.
+
+### 7. Common Issues and Troubleshooting:
+
+#### 7.1. Resource Issues:
+
+##### Error Message:
+```python
+24/06/16 00:55:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
+```
+
+This warning indicates that the Spark job could not acquire the necessary resources to start execution. Please contact
+the CDM team for assistance in resolving this issue.
+
+#### 7.2 SparkMonitor Issues:
+
+##### Error Message:
+```python
+Exception in thread Thread-5:
+Traceback (most recent call last):
+  File "/opt/bitnami/python/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
+    self.run()
+  File "/opt/bitnami/python/lib/python3.11/site-packages/sparkmonitor/kernelextension.py", line 126, in run
+    self.onrecv(msg)
+  File "/opt/bitnami/python/lib/python3.11/site-packages/sparkmonitor/kernelextension.py", line 143, in onrecv
+    sendToFrontEnd({
+  File "/opt/bitnami/python/lib/python3.11/site-packages/sparkmonitor/kernelextension.py", line 223, in sendToFrontEnd
+    monitor.send(msg)
+  File "/opt/bitnami/python/lib/python3.11/site-packages/sparkmonitor/kernelextension.py", line 57, in send
+    self.comm.send(msg)
+    ^^^^^^^^^
+AttributeError: 'ScalaMonitor' object has no attribute 'comm'
+```
+
+This error occurs when there is an issue with the SparkMonitor kernel extension, specifically related to the 
+communication between the kernel and the front end. The error indicates that the comm attribute is missing from the 
+ScalaMonitor object.
```

Comment:
What causes this?

Reply:
From the extension developer:

`This is a difficult error to debug; the kernel extension is complaining that the Scala Listener has no comms; this is usually because the listener wasn't able to find a Spark Instance.`
https://github.com/itsjafer/jupyterlab-sparkmonitor/issues/16

Reply:
Hmm, that ticket says the bug is fixed

Reply:
yea. It didn't happen very often. And restart kernel always works.